### PR TITLE
feat: new sidebar

### PR DESCRIPTION
**Pull request by scmmishra**
This PR has the initial version of the new sidebar targeted for the next major redesign of the app. This PR includes the following changes

- Components in the `layouts-next` and `base-next` directories in `dashboard/components`
- Two generic components `Avatar` and `Icon`
- `SidebarGroup` component to manage expandable sidebar groups with nested navigation items. This includes handling active states, transitions, and permissions.
- `SidebarGroupHeader` component to display the header of each navigation group with optional icons and active state indication.
- `SidebarGroupLeaf` component for individual navigation items within a group, supporting icons and active state.
- `SidebarGroupSeparator` component to visually separate nested navigation items. (They look a lot like header)
- `SidebarGroupEmptyLeaf` component to render empty state of any navigation groups.

CleanShot 2024-10-18 at 21 31 33@2x

### Pending Tasks
- [x] Figure out portal navigation, since it needs an active portal (Needs discussion)
- [x] Figure out how do I find empty sub-groups if none of the children are visible
- [x] Figure out navigation when clicking the group header, right now the behaviour is not clear. (Needs discussion)

### CODE REVIEW

File: app/javascript/dashboard/components-next/sidebar/SidebarGroupEmptyLeaf.vue [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<template>
+  <li
+    class="py-1 pl-3 text-n-slate-10 border rounded-lg border-dashed text-center border-n-alpha-2 text-xs h-8 grid place-content-center select-none pointer-events-none"
+  >
+    {{ 'No items' }}
```
Comment (pranavrajs) [inline app/javascript/dashboard/components-next/sidebar/SidebarGroupEmptyLeaf.vue:5]: i18n

File: config/features.yml [hunk 2]
```diff
@@ -92,3 +92,5 @@
 - name: custom_roles
   enabled: false
   premium: true
+- name: sidebar_next
```
Comment (pranavrajs) [inline config/features.yml:95]:
```suggestion
- name: chatwoot_v4
```

Comment (pranavrajs) [reply config/features.yml:95]: Let us use only one flag for every item.

File: app/javascript/dashboard/components-next/sidebar/ChannelLeaf.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+const reauthorizationRequired = computed(() => {
+  return props.inbox.reauthorization_required;
+});
+</script>
+
+<template>
+  <span
+    class="size-4 grid place-content-center rounded-full bg-n-alpha-2"
+    :class="{ 'bg-n-blue/20': active }"
```
Comment (pranavrajs) [inline app/javascript/dashboard/components-next/sidebar/ChannelLeaf.vue:59]: Do we have blue defined in the palette?

File: app/javascript/dashboard/components-next/sidebar/Sidebar.vue [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+        />
+      </div>
+      <div class="gap-2 flex px-2">
+        <RouterLink
+          :to="{ name: 'search' }"
+          class="rounded-lg py-1 flex items-center gap-2 px-2 border-n-weak border bg-n-solid-3 dark:bg-n-black/30 w-full"
+        >
+          <span class="i-lucide-search size-4 text-n-slate-11 flex-shrink-0" />
+          <span class="flex-grow text-left">
+            {{ 'Search...' }}
```
Comment (pranavrajs) [inline app/javascript/dashboard/components-next/sidebar/Sidebar.vue:424]: Please use i18n everywhere.

File: app/javascript/dashboard/components-next/sidebar/SidebarProfileMenu.vue [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  return menuItems.value.filter(item => item.show);
+});
+</script>
+
+<template>
+  <div class="relative z-20 w-full min-w-0">
+    <button
+      class="flex gap-2 items-center rounded-lg cursor-pointer text-left w-full"
+      v-bind="$attrs"
+      @click="toggleProfileMenu()"
```
Comment (pranavrajs) [inline app/javascript/dashboard/components-next/sidebar/SidebarProfileMenu.vue:115]:
```suggestion
      @click="toggleProfileMenu"
```

File: app/javascript/dashboard/components-next/sidebar/ChannelLeaf.vue (hunk 3 above)
Comment (scmmishra) [reply app/javascript/dashboard/components-next/sidebar/ChannelLeaf.vue:59]: Yep! It is. CC @absurdiya 

CleanShot 2024-10-23 at 12 48 29@2x

File: app/javascript/dashboard/components-next/sidebar/Sidebar.vue [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+        />
+      </ul>
+    </nav>
+    <section
+      class="p-1 border-t border-n-strong shadow-[0px_-2px_4px_0px_rgba(27,28,29,0.02)] flex-shrink-0 flex justify-between gap-2 items-center"
+    >
+      <SidebarProfileMenu
+        @open-key-shortcut-modal="emit('openKeyShortcutModal')"
+      />
+      <div v-if="false" class="flex items-center">
```
Comment (pranavrajs) [inline app/javascript/dashboard/components-next/sidebar/Sidebar.vue:465]: I hope this is still kept for future updates.

File: app/javascript/dashboard/components/layout/Sidebar.vue [hunk 7]
```diff
@@ -104,6 +104,9 @@ export default {
       return '';
     },
     customViews() {
+      if (!this.activeCustomView) {
```
Comment (pranavrajs) [inline app/javascript/dashboard/components/layout/Sidebar.vue:107]: Added this as it was breaking the Sidebar (old) when moving from Conversations to Settings pages.

**Review by pranavrajs — APPROVED** [review]

**Comment by KokeroO** [comment]
@pranavrajs @scmmishra Will the notification bell be removed permanently or has it been forgotten?

**Comment by pranavrajs** [comment]
@KokeroO This change is intentional. The notification bell will be removed and replaced with the inbox view, which you can test by enabling the inbox_view feature flag. This new setup offers greater control over notifications, with options to snooze or delete them, along with improved sorting features.

**Pull request merged**